### DISCUSSION TITLE
Api Gateway : Action is returning result - looks like Operation can't passThrough

### DISCUSSION THREAD

**Discussion by marifrahman** [apioo/fusio#539] [Q&A] [answered]
I have a api gateway created for Bard - looks like Action is returning the result. However, when I access the Operation endpoint from POSTMAN I get an error:
`Error: Parse Error: Invalid character in chunk size`

My Action looks like this : 
<img width="587" alt="image" src="https://github.com/apioo/fusio/assets/772407/13960e75-c5fc-4366-97dc-ddeab6f3126c">

And the Operation looks like this:
<img width="596" alt="image" src="https://github.com/apioo/fusio/assets/772407/8e92fce9-1b02-449d-a579-2591e116bdec">

Any idea what's going on ? Anything to with the header from the external api vs api from fusio ? Anyway to debug ?

**Comment by chriskapp** [comment]
Hi @marifrahman could you manually call the endpoint and show me the complete HTTP response header and body maybe this is an error which gets returned by the endpoint and Fusio tries to JSON decode this response.

**Reply by marifrahman** [reply]
Much appreciated your response @chriskapp.
Here is the response of the manual call for the endpoint from POSTMAN : 
<img width="614" alt="image" src="https://github.com/apioo/fusio/assets/772407/df136075-c9c6-4288-85c7-2fd8fcc27515">
Response body : 
`{
  "candidates": [
    {
      "output": "Fine.",
      "safetyRatings": [
        {
          "category": "HARM_CATEGORY_DEROGATORY",
          "probability": "NEGLIGIBLE"
        },
        {
          "category": "HARM_CATEGORY_TOXICITY",
          "probability": "NEGLIGIBLE"
        },
        {
          "category": "HARM_CATEGORY_VIOLENCE",
          "probability": "NEGLIGIBLE"
        },
        {
          "category": "HARM_CATEGORY_SEXUAL",
          "probability": "NEGLIGIBLE"
        },
        {
          "category": "HARM_CATEGORY_MEDICAL",
          "probability": "NEGLIGIBLE"
        },
        {
          "category": "HARM_CATEGORY_DANGEROUS",
          "probability": "NEGLIGIBLE"
        }
      ]
    }
  ]
}
`
Response header: 
<img width="601" alt="image" src="https://github.com/apioo/fusio/assets/772407/f286f774-6d4f-4aee-8732-131f9621a072">

From the Action designer when I execute - I get the result from there as well : 
<img width="881" alt="image" src="https://github.com/apioo/fusio/assets/772407/b9d5e708-48f0-442d-90b9-e8686f80578a">

So I am assuming something to do in my Operation where the result from the endpoint is be passThrough ?

**Reply by marifrahman** [reply]
Hi,
Just checking on it again - if anyone got any clue on this issue ? Is there anything to do with the delay on the webservice ?

**Reply by chriskapp** [reply, marked as answer]
@marifrahman ok I think I understand the problem, it is probably because the API returns the response with a "Transfer-Encoding" header set to "chunked". To test this, could you change the file `vendor/fusio/adapter-http/src/Action/HttpSenderAbstract.php` with the following fix s.
![image](https://github.com/apioo/fusio/assets/2505846/d228eee9-0fbb-4398-a61a-f9d2f7371832)

**Reply by marifrahman** [reply]
Awesome that is the case. Should be a fix then ?

On another note: it looks like when passThru is used for incoming schema - looks like parameters set at the operations are Not passed to the action but only the parameters coming from the client. Is this the case or not sure if I am doing something wrong !
<img width="562" alt="image" src="https://github.com/apioo/fusio/assets/772407/408d3028-7aa6-43e8-8d33-e0cc426f9f9f">

**Reply by chriskapp** [reply]
ok, great the fix regarding `Transfer-Encoding` will be included in the next release. Regarding your last point, the field where you have provided the actual value is a "description" field, there you can only describe the parameter, this description is then also used at the OpenAPI specification, so it is not the place where you can provide an actual value. But at the "bardProxy" action it should be possible to provide a default parameter value, depending on the action which you use.

**Reply by marifrahman** [reply]
Got it. Thanks again.
I am using HTTP-Processor action and trying to pass the parameter as part of uri as bellow; doesn't work.
![image](https://github.com/apioo/fusio/assets/772407/f7f73d48-c2fb-476f-924c-bff377bfffeb)

However, from Action designer when I call it with parameter set - it does work !!
![image](https://github.com/apioo/fusio/assets/772407/4c25ee9c-80b7-4d0d-9541-0f85e4b2f009)

**Reply by chriskapp** [reply]
ok, providing the query parameters in the url will indeed not work, there is a new "Query" config option at the latest HTTP Processor action s. https://github.com/apioo/fusio-adapter-http/blob/master/src/Action/HttpProcessor.php#L67
which will be available in the next version. If you want to test this you could run
```
composer update fusio/adapter-http
```
to get the latest version of the HTTP adapter, since the version is already released, otherwise you can of course also wait for the next release, which should be available soon.
